### ISSUE TITLE
Questions about gradient and ism script

### ISSUE THREAD

**Issue by FakeEnd** [johahi/borzoi-pytorch#24]
Hi, thanks for your nice repo!

I try to reproduce the gradient and ism script in crispr experiment using pytorch, is there any reference code you suggested?

Thanks very much

**Comment by johahi** [comment]
Hi, 
sorry for the delay, here's a snippet of how we scored the encode e2g enhancers:
```python
for batch in tqdm.tqdm(crispr_dl,miniters = 10):
    seq, bins, enh_mid = batch
    seq = seq.to(device)
    seq = seq.permute(0,2,1)
    seq.requires_grad = True
    with torch.autocast(device, enabled=not disable_autocast):
        # predict gene coverage
        pred = borzoi.predict_gene_count(seq, [bins.squeeze()])
        # average over relevant tracks
        pred = pred.mean(axis=-1) 
        #pred = torch.log(pred + 1)
        # compute gradient
    pred.backward()
    grads = seq.grad.clone()
    # remove mean across nucleotides 
    grads = grads - grads.mean(axis=1, keepdims=True)
    # compute absolute reference gradient
    grads = torch.abs(grads*seq).amax(axis=1).squeeze()
    # average over region and normalize
    enh_grad = grads[enh_mid - 1000:enh_mid + 1000].mean()
    enh_grad = enh_grad - grads.mean()
    # return 
    preds.append(enh_grad.detach().cpu())
```
Does that help?

**Comment by FakeEnd** [comment]
Yes, this is very helpful — thank you so much for sharing the snippet.

I also wanted to ask about `crispr_dl` in your code. Is there a reference implementation or dataset loader for it available somewhere?

Thanks again!
